reactive/apache.py

```diff
@@ -94,6 +94,11 @@ def strip_archive_dir(site_dir):
     subdir = os.path.join(site_dir, contents[0])
     if not os.path.isdir(subdir):
         return
+    # check for the case where subdir have the same name of the parent
+    if os.path.basename(site_dir) == contents[0]:
+        os.rename(subdir, subdir+"_tmp")
+        contents = os.listdir(site_dir)
+        subdir = os.path.join(site_dir, contents[0])
```

Comment:
These two lines seem a little confusing at first glance.  I think it would be a bit easier to scan as:

```python
os.rename(subdir, subdir + '_tmp')
subdir = subdir + '_tmp'
```

Reply:
Ok, I'll work on that

Reply:
It's interesting how a thing so simple can get confusing if the variable names are not appropriate. I re-wrote a bigger part, please tell me what you think of this:

```suggestion
def strip_archive_dir(site_dir):
    contents = os.listdir(site_dir)
    if len(contents) != 1:
        return
    archive_topdir = contents[0]
    archive_topdir_fullpath = os.path.join(site_dir, archive_topdir)
    if not os.path.isdir(archive_topdir_fullpath):
        return
    # check for the case where archive top dir have the same name of the parent dir (site name)
    if os.path.basename(site_dir) == archive_topdir:
        os.rename(archive_topdir_fullpath, archive_topdir_fullpath + "_tmp")
        archive_topdir += "_tmp"
        archive_topdir_fullpath += "_tmp"
    tmp_dest = os.path.join(os.path.dirname(site_dir), archive_topdir)
    os.rename(archive_topdir_fullpath, tmp_dest)
    os.rmdir(site_dir)
    os.rename(tmp_dest, site_dir)
```

Reply:
If you like I will submit another proper pull request. Feel free to dislike or suggest more changes.

Reply:
I think that's quite a bit clearer, but the better understanding that gave me made me think of something that I think is even better:

```python
def strip_archive_dir(site_dir):
    contents = os.listdir(site_dir)
    if len(contents) != 1:
        return
    archive_topdir = contents[0]
    archive_topdir_fullpath = os.path.join(site_dir, archive_topdir)
    if not os.path.isdir(archive_topdir_fullpath):
        return
    tmp_dest = tempdir.mkdtemp(dir=os.path.dirname(site_dir))
    os.rename(archive_topdir_fullpath, tmp_dest)
    os.rmdir(site_dir)
    os.rename(tmp_dest, site_dir)
```

Thoughts?

Reply:
yes! much cleaner, I should have thought about that, hehe. Do you want me to submit that or you do it?

Reply:
If you're ok with that, and don't mind confirming that it solves your use-case and updating the PR, I can merge it here.

Reply:
Thank you!

Reply:
I'm testing and will report back that it works

Reply:
I changed some more bits, please check. I push --force'd the last changes to the same repo, I hope the pull request will update accordingly. This pushed version is confirmed working.